### ISSUE TITLE
Remove --host and --port in favour of --bind-addr

### ISSUE THREAD

**Issue by nhooyr** [cdr/code-server#1013]
More ergonomic and in line with other modern tools.

E.g. `--bind-addr=localhost:8080`

**Comment by sr229** [comment]
`bind-addr` has more keystrokes than `address` so I think we should go for `address` instead.

**Comment by nhooyr** [comment]
Address isn't clear on what it is. Is it the external address at which the code-server is available? bind-addr makes it very clear and there is precedent for it in a lot of tooling I've used before.

**Comment by antofthy** [comment]
I also want to see `--bind-address` implemented as separate to `--host`  as they are NOT the same

If I give `--host  machine.some.domain`  code server will then display a URL with the resolved IP
of that hostname.   On our systems that IP is the  internal Docker subnet IP, and not the IP the user will use.   Our users need to us the 'hostname' as this points to a wildcard proxy to a docker swarm.

Basically there are two host names involved...
       1/  binding host/IP
       2/  hostname used for redirects and reporting to user

The second could be made part of `--base-path`  though it willmake it more of a **--base-url**
However my attempts to set a --base-path has resulted in "error File not found"

**Comment by nhooyr** [comment]
Closed by https://github.com/cdr/code-server/pull/1562

**Comment by antofthy** [comment]
So exactly what changed? Just fixing it is no good if the users don't know what changed!

**Comment by nhooyr** [comment]
> If I give --host machine.some.domain code server will then display a URL with the resolved IP
of that hostname. On our systems that IP is the internal Docker subnet IP, and not the IP the user will use. Our users need to use the given argument as this points to a wildcard proxy to a docker swarm.

I'm not I understand what you mean exactly. Where will code-server display this URL and why?
